main.nf

```diff
@@ -268,114 +200,377 @@ ${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style
 
+    script:
+    """
+    fastqc -t "${task.cpus}" -q ${reads}
+    """
+}
 
-=======
 
 /*
- * STEP 1 - FastQC
+ * STEP 2 - Assembly
  */
-process fastqc {
+process megahit {
     tag "$name"
-    publishDir "${params.outdir}/fastqc", mode: 'copy',
-        saveAs: {filename -> filename.indexOf(".zip") > 0 ? "zips/$filename" : "$filename"}
+    publishDir "${params.outdir}/", mode: 'copy'
 
     input:
-    set val(name), file(reads) from read_files_fastqc
+    set val(name), file(reads) from trimmed_reads_megahit
 
     output:
-    file "*_fastqc.{zip,html}" into fastqc_results
+    set val(name), file("megahit/${name}.contigs.fa") into megahit_assembly
+
+    script:
+    if ( !params.singleEnd ) {
+    """
+    megahit -t "${task.cpus}" -1 "${reads[0]}" -2 "${reads[1]}" -o megahit \
+        --out-prefix "${name}"
+    """
+    }
+    else {
+    """
+    megahit -t "${task.cpus}" -r ${reads} -o megahit --out-prefix "${name}"
+    """
+    }
+}
+megahit_assembly.into{assembly_quast; assembly_metabat; assembly_refinem}
+
+
+
+process quast {
+    tag "$name"
+    // publishDir "${params.outdir}/quast/$name", mode: 'copy'
+
+    input:
+    set val(name), file(assembly) from assembly_quast
+
+    output:
+    file("quast_${name}/*") into quast_results
+
+    script:
+    """
+    quast.py --threads "${task.cpus}" -l "${name}" "${assembly}" -o "quast_${name}"
+    """
+}
+
+
+/*
+ * STEP 3 - Binning
+ */
+process metabat {
+    tag "$name"
+    publishDir "${params.outdir}/metabat", mode: 'copy'
+
+    input:
+    set val(_name), file(assembly) from assembly_metabat
+    set val(name), file(reads) from trimmed_reads_metabat
+    val(min_size) from params.min_contig_size
+
+    output:
+    set val(name), file("bins/") into metabat_bins
+    set val(name), file("${name}.bam") into mapped_reads_assembly
 
     script:
+    if ( !params.singleEnd ) {
     """
-    fastqc -q $reads
+    bowtie2-build --threads "${task.cpus}" "${assembly}" ref
+    bowtie2 -p "${task.cpus}" -x ref -1 "${reads[0]}" -2 "${reads[1]}" | \
+        samtools view -@ "${task.cpus}" -bS | \
+        samtools sort -@ "${task.cpus}" -o "${name}.bam"
+    samtools index "${name}.bam"
+    jgi_summarize_bam_contig_depths --outputDepth depth.txt "${name}.bam"
+    metabat2 -t "${task.cpus}" -i "${assembly}" -a depth.txt -o "bins/${name}" -m ${min_size}
     """
+    }
+    else {
+    """
+    bowtie2-build --threads "${task.cpus}" "${assembly}" ref
+    bowtie2 -p "${task.cpus}" -x ref -U ${reads} | \
+        samtools view -@ "${task.cpus}" -bS | \
+        samtools sort -@ "${task.cpus}" -o "${name}.bam"
+    samtools index "${name}.bam"
+    jgi_summarize_bam_contig_depths --outputDepth depth.txt "${name}.bam"
+    metabat2 -t "${task.cpus}" -i "${assembly}" -a depth.txt -o bins/"${name}" -m ${min_size}
+    """
+    }
+
 }
 
+mapped_reads_assembly.into{mapped_reads_checkm; mapped_reads_refinem}
 
+process checkm_download_db {
+    time '4h'
```

Comment:
Should be done in the `base.config` instead, so that everyone can benefit from it :-)